src/topology/instances/add_circle.lean

```diff
@@ -126,6 +154,86 @@ instance : divisible_by (add_circle p) ℤ :=
     exact (equiv_Ico p).symm_apply_apply x,
   end, }
 
+end floor_ring
+
+section finite_order_points
+
+variables {p}
+
+/-- See also `add_circle.gcd_mul_add_order_of_div_eq`. -/
+lemma add_order_of_div_of_gcd_eq_one {m n : ℕ} (hn : 0 < n) (h : gcd n m = 1) :
+  add_order_of (↑(↑m / ↑n * p) : add_circle p) = n :=
+begin
+  rcases m.eq_zero_or_pos with rfl | hm, { rw [gcd_zero_right, normalize_eq] at h, simp [h], },
+  let x : add_circle p := ↑(↑m / ↑n * p),
```

Comment:
not required, but I think this makes the goal more readable
```suggestion
  set x : add_circle p := ↑(↑m / ↑n * p),
```

Reply:
OK done.